src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -410,52 +411,41 @@ export class ActivityGraphFillComputer {
     const afterSampleCpuRatio =
       cpuAfterSample === null ? 1 : cpuAfterSample / maxThreadCPUDelta;
 
-    // Samples have two parts to be able to present the CPU utilizations properly.
-    // The first half of the sample will use the CPU delta number that belongs to
-    // this sample.
+    // Samples have two parts to be able to present the different CPU usages properly.
+    // This is because CPU usage number of a sample represents the CPU usage
+    // starting starting from the previous sample time to this sample time.
+    // These parts will be:
+    // - Between `sampleCategoryStartPixel` and `samplePixel` with beforeSampleCpuRatio.
+    // - Between `samplePixel` and `sampleCategoryEndPixel` with afterSampleCpuRatio.
+
+    // Here we are accumulating the first part of the sample. It will use the
+    // CPU delta number that belongs to this sample.
+    // This part starts from the "sample start time" to "sample time" and uses
+    // beforeSampleCpuRatio.
     for (let i = intCategoryStartPixel; i <= intSamplePixel; i++) {
       percentageBuffer[i] += beforeSampleCpuRatio;
     }
-    // The second half of the sample will use the CPU delta number that belongs to
-    // the next sample.
-    // For the samples that are consist of only one pixel, this loop will not be
-    // executed. It will only be executed if sample has 2 or more pixels.
-    for (let i = intSamplePixel + 1; i <= intCategoryEndPixel; i++) {
```

Comment:
Previously it was starting from `intSamplePixel + 1` to not deal with the subpixel subtraction in between. But we don't need this anymore since we handle the subtraction now properly.